base/essentials.jl

```diff
@@ -447,6 +448,9 @@ Stacktrace:
 ```
 """
 sizeof(x) = Core.sizeof(x)
+# The next two methods prevent invalidation
+sizeof(::Type{Union{}}) = Core.sizeof(Union{})
+sizeof(::Type{T}) where T = Core.sizeof(T)
```

Comment:
This'll force us to generate code for each type, and forces us to add a dispatch, which I believe we don't want for it?